src/Http/Ownership/SuggestOwnershipsRequestHandler.php

```diff
@@ -0,0 +1,148 @@
+{
+    private ResultsGeneratorInterface $resultsGenerator;
+    private OfferJsonDocumentReadRepository $offerRepository;
+    private CurrentUser $currentUser;
+    private UserIdentityResolver $userIdentityResolver;
+    private DocumentRepository $organizerRepository;
+    private OwnershipSearchRepository $ownershipSearchRepository;
+    private OrganizerIDParser $organizerIDParser;
+    private UserPermissionsReadRepositoryInterface $userPermissionsRepository;
+
+    public function __construct(
+        SearchServiceInterface $searchService,
+        OfferJsonDocumentReadRepository $offerRepository,
+        CurrentUser $currentUser,
+        UserIdentityResolver $userIdentityResolver,
+        DocumentRepository $organizerRepository,
+        OwnershipSearchRepository $ownershipSearchRepository,
+        OrganizerIDParser $organizerIDParser,
+        UserPermissionsReadRepositoryInterface $userPermissionsRepository
+    ) {
+        $this->resultsGenerator = new ResultsGenerator(
+            $searchService,
+            new Sorting('modified', 'desc')
+        );
+        $this->offerRepository = $offerRepository;
+        $this->currentUser = $currentUser;
+        $this->userIdentityResolver = $userIdentityResolver;
+        $this->organizerRepository = $organizerRepository;
+        $this->ownershipSearchRepository = $ownershipSearchRepository;
+        $this->organizerIDParser = $organizerIDParser;
+        $this->userPermissionsRepository = $userPermissionsRepository;
+    }
+
+    public function handle(ServerRequestInterface $request): ResponseInterface
+    {
+        $this->guardAccess();
+
+        $queryParams = new QueryParameters($request);
+
+        $queryParams->guardRequiredEnum('itemType', [ItemType::organizer()->toString()]);
+
+        $id = $this->currentUser->getId();
+        $user = $this->userIdentityResolver->getUserById($id);
+
+        $query = (new SuggestOwnershipsSapiQuery($user))->toString();
+
+        $activeOwnerships = $this->ownershipSearchRepository
+            ->search(new SearchQuery([
+                new SearchParameter('ownerId', $user->getUserId()),
+            ]))
+            ->filter(self::activeOwnerships());
+
+        /**
+         * A map to deduplicate returned organizers
+         * @var array<string, array> $idToOrganizerMap
+         */
+        $idToOrganizerMap = [];
+
+        /**
+         * @var ItemIdentifier $result
+         */
+        foreach ($this->resultsGenerator->search($query) as $result) {
+            $offerType = OfferType::fromCaseInsensitiveValue($result->getItemType()->toString());
+            $id = $result->getId();
+
+            $offer = $this->offerRepository->fetch($offerType, $id)->getAssocBody();
+
+            $organizerId = $this->organizerIDParser->fromUrl(new Url($offer['organizer']['@id']))->toString();
+
+            $ownershipsForOrganizer = $activeOwnerships->filter(fn (OwnershipItem $item) => $item->getItemId() === $organizerId);
```

Comment:
Is it not more efficient to just get all the organizer IDs first and then filter out the once with already an ownership? That would limit the number of filter calls.